EmotiBitEda.cpp

```diff
@@ -28,32 +28,539 @@
 /**************************************************************************/
 
 #include "EmotiBitEda.h"
+#include "EmotiBitEdaCalibration.h"
 
-/**************************************************************************/
-/*!
-    @brief  Sets EmotiBit to read EDA data
+bool EmotiBitEda::setup(EmotiBitVersionController::EmotiBitVersion version, float samplingRate,
+	DoubleBufferFloat* edaBuffer, DoubleBufferFloat* edlBuffer, DoubleBufferFloat* edrBuffer,
+	TwoWire* emotibitI2c, BufferFloat* edlOversampBuffer, BufferFloat* edrOversampBuffer)
+{
+	_emotibitVersion = version;
+	_constants.samplingRate = samplingRate;
 
-    @param version of EmotiBit
-    @param emotibit_i2c I2C bus
+	// ToDo: Calculate digFiltAlpha
 
-    @return true if successful, otherwise false
-*/
-/**************************************************************************/
-bool EmotiBitEda::setup(EmotiBitVersionController::EmotiBitVersion version, TwoWire* emotibit_i2c)
-{
-	// ToDo: add sampling rate inputs
-	
-	if (version == EmotiBitVersionController::EmotiBitVersion::V04A)
+	_edaBuffer = edaBuffer;
+	_edlBuffer = edlBuffer;
+	_edrBuffer = edrBuffer;
+	_edlOversampBuffer = edlOversampBuffer;
+	_edrOversampBuffer = edrOversampBuffer;
+
+	String output;
+	output.reserve(30);
+
+	output = "edaSeriesResistance: " + String(_constants.edaSeriesResistance);
+	Serial.println(output);
+	output = "samplingRate: " + String(_constants.samplingRate);
+	Serial.println(output);
+
+	if (_emotibitVersion >= EmotiBitVersionController::EmotiBitVersion::V04A)
 	{	
-		ads.setDataRate(RATE_ADS1115_475SPS);	// set to 475Hz to allow for 300Hz oversampling
-		ads.setGain(GAIN_TWO);        // 2x gain   +/- 2.048V  1 bit = 1mV      0.0625mV
-		return ads.begin(0x48, emotibit_i2c, false); // callBegin -> false. the i2c wire has already been init in setup
+		output = "Configuring ADS ADC... ";
+		Serial.println(output);
+
+		_constants.adcBits = 16;
+
+		// NOTE: if these values are changed in code, we should add parameters to _info.json
+		_ads.setDataRate(RATE_ADS1115_475SPS);	// set to 475Hz to allow for 300Hz oversampling
+		_ads.setGain(GAIN_TWO);        // 2x gain   +/- 2.048V  1 bit = 1mV      0.0625mV
+
+		_constants.clipMin = -26500;
+		_constants.clipMax = 26500;
```

Comment:
The EmotiBit I was using was giving a value of ~26433 with an open circuit. And, even with an open circuit, it was not generating any clipping events because the range specified here 26500(which allows for an open circuit to exist without clipping). Shouldn't we be generating clipping events for open and short circuits?